File: examples/compile_wat.py
#!/usr/bin/env python3
"""
Compile WAT to WASM using Python's wasmtime package
Install with: pip install wasmtime
"""
import sys

try:
    from wasmtime import wat2wasm
except ImportError:
    print("Error: wasmtime package not installed")
    print("Install with: pip install wasmtime")
    sys.exit(1)

# Read the WAT file
with open('examples/hello.wat', 'r') as f:
    wat_content = f.read()

# Convert to WASM
wasm_bytes = wat2wasm(wat_content)

# Write the WASM file
with open('examples/hello.wasm', 'wb') as f:
    f.write(wasm_bytes)

print(f"Successfully compiled hello.wat to hello.wasm ({len(wasm_bytes)} bytes)")

File: examples/compile_wat.rs
use std::fs;

fn main() -> anyhow::Result<()> {
    // Read the WAT file
    let wat = fs::read_to_string("examples/hello.wat")?;

    // Convert to binary
    let wasm = wasmtime::wat::parse_str(&wat)?;

    // Write the binary
    fs::write("examples/hello.wasm", wasm)?;

    println!("Successfully compiled hello.wat to hello.wasm");
    Ok(())
}
